### ISSUE TITLE
Improve search function accessibility

### ISSUE THREAD

**Issue by crunchyruby** [crunchyruby/learning-notes#11]
# Enhancement: Improve search function accessibility

## Current Situation
Currently, the search function's accessibility is missing some useful features.

## Proposed Improvement
We should improve it by adding more options.

## Benefits
- Better user experience when using the search function
- Improved accessibility across the application

## Implementation Ideas
- Refactor the search function code to be more modular
- Add new accessibility-focused features to the search function
